core/client/db/src/cache/mod.rs

```diff
@@ -134,21 +157,27 @@ impl<'a, Block: BlockT> DbCacheTransaction<'a, Block> {
 		mut self,
 		parent: ComplexBlockId<Block>,
 		block: ComplexBlockId<Block>,
-		authorities_at: Option<Vec<AuthorityIdFor<Block>>>,
+		data_at: HashMap<Vec<u8>, Vec<u8>>,
```

Comment:
Could this be changed to a more general `Iterator`?

Reply:
@DemiMarie-parity Probably, yes, but I'm not sure if it makes the API better. I suppose that working with the cache as `HashMap` is more natural.

@svyatonik What do you think?

Reply:
@DarkEld3r I agree that using `HashMap` here would be better :)

Reply:
@DarkEld3r @svyatonik I agree that using `HashMap` is better.  For starters, it reduces the need for future, backwards-incompatible changes.